engine/src/main/java/org/pentaho/di/job/entries/exportrepository/JobEntryExportRepository.java

```diff
@@ -0,0 +1,826 @@
+    try {
+      if ( !repdir.isRoot() ) {
+        if ( repdir.toString().lastIndexOf( "/" ) == 0 ) {
+          String filename = repdir.toString().replace( "/", "" );
+          String foldername = realoutfilename;
+          if ( newfolder ) {
+            foldername = realoutfilename + Const.FILE_SEPARATOR + filename;
+            this.file = KettleVFS.getFileObject( foldername, this );
+            if ( !this.file.exists() ) {
+              this.file.createFolder();
+            }
+          }
+
+          filename = foldername + Const.FILE_SEPARATOR + buildFilename( filename ) + ".xml";
+          this.file = KettleVFS.getFileObject( filename, this );
+
+          if ( this.file.exists() ) {
+            if ( iffileexists.equals( If_FileExists_Skip ) ) {
+              // Skip this folder
+              return true;
+            } else if ( iffileexists.equals( If_FileExists_Uniquename ) ) {
+              filename = realoutfilename + Const.FILE_SEPARATOR + buildUniqueFilename( filename ) + ".xml";
+            } else if ( iffileexists.equals( If_FileExists_Fail ) ) {
+              // Fail
+              return false;
+            }
+          }
+
+          // System.out.print(filename + "\n");
+          if ( log.isDetailed() ) {
+            logDetailed( "---" );
+            logDetailed( BaseMessages.getString(
+              PKG, "JobExportRepository.Log.FolderProcessing", "" + folderno, "" + totalfolders ) );
+            logDetailed( BaseMessages.getString(
+              PKG, "JobExportRepository.Log.OutFilename", repdir.toString(), filename ) );
+          }
+
+          new RepositoryExporter( this.repository ).exportAllObjects( null, filename, repdir, "all" );
+          if ( log.isDetailed() ) {
+            logDetailed( BaseMessages.getString(
+              PKG, "JobExportRepository.Log.OutFilenameEnd", repdir.toString(), filename ) );
+          }
+
+          if ( add_result_filesname ) {
+            addFileToResultFilenames( filename, log, result, parentJob );
+          }
+
+        }
+      } // end if root
+      retval = true;
+    } catch ( Exception e ) {
+      // Update errors
+      updateErrors();
+      logError( BaseMessages.getString( PKG, "JobExportRepository.ErrorExportingFolder", repdir.toString(), e
+        .toString() ) );
+    }
+    return retval;
+  }
+
+  private boolean checkIfSuccessConditionBroken() {
+    boolean retval = false;
+    if ( ( this.NrErrors > 0 && getSuccessCondition().equals( SUCCESS_IF_NO_ERRORS ) )
+      || ( this.NrErrors >= this.limitErr && getSuccessCondition().equals( SUCCESS_IF_ERRORS_LESS ) ) ) {
+      retval = true;
+    }
+    return retval;
+  }
+
+  private void connectRep( LogChannelInterface log, String realrepName, String realusername, String realpassword ) throws Exception {
+    this.repsinfo = new RepositoriesMeta();
+    this.repsinfo.getLog().setLogLevel( log.getLogLevel() );
+    try {
+      this.repsinfo.readData();
+    } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)